inference-engine/src/transformations/src/transformations/common_optimizations/concat_reduce_fusion.cpp

```diff
@@ -0,0 +1,114 @@
+// Copyright (C) 2021 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "transformations/common_optimizations/concat_reduce_fusion.hpp"
+#include "transformations/utils/utils.hpp"
+
+#include <memory>
+#include <vector>
+
+#include <ngraph/opsets/opset8.hpp>
+#include <ngraph/rt_info.hpp>
+#include <ngraph/pattern/op/wrap_type.hpp>
+#include "itt.hpp"
+
+
+NGRAPH_RTTI_DEFINITION(ngraph::pass::ConcatReduceMaxFusion, "ConcatReduceMaxFusion", 0);
+NGRAPH_RTTI_DEFINITION(ngraph::pass::ConcatReduceMinFusion, "ConcatReduceMinFusion", 0);
+NGRAPH_RTTI_DEFINITION(ngraph::pass::ConcatReduceFusion, "ConcatReduceFusion", 0);
+
+
+ngraph::pass::ConcatReduceMaxFusion::ConcatReduceMaxFusion() {
+    MATCHER_SCOPE(ConcatReduceMaxFusion);
+
+    auto concat_pattern = ngraph::pattern::wrap_type<opset8::Concat>({pattern::any_input(), pattern::any_input()});
+    auto reduce_axes_pattern = ngraph::pattern::wrap_type<opset8::Constant>();
+    auto reduce_pattern = ngraph::pattern::wrap_type<opset8::ReduceMax>({concat_pattern, reduce_axes_pattern});
+
+    ngraph::matcher_pass_callback callback = [=](pattern::Matcher& m) {
+        const auto& pattern_map = m.get_pattern_value_map();
+
+        auto concat = std::dynamic_pointer_cast<opset8::Concat>(pattern_map.at(concat_pattern).get_node_shared_ptr());
```

Comment:
You need to check casted value for nullptr.
```suggestion
        auto concat = std::dynamic_pointer_cast<opset8::Concat>(pattern_map.at(concat_pattern).get_node_shared_ptr());
         auto reduce = std::dynamic_pointer_cast<opset8::ReduceMax>(pattern_map.at(reduce_pattern).get_node_shared_ptr());
         if (!concat || !reduce) {
             return false;
         }
```